### PR TITLE
store: fix download caching and add integration test

### PR DESCRIPTION
**Pull request by mvo5**
The download caching in the store got recently broken in PR#4075
because of insufficient integration tests.

This PR re-adds the store download caching and adds a spread
integration test to ensure this won't happen again.

### CODE REVIEW

File: overlord/overlord.go [hunk 1]
```diff
@@ -142,6 +142,7 @@ func New() (*Overlord, error) {
 	// setting up the store
 	authContext := auth.NewAuthContext(s, o.deviceMgr)
 	sto := storeNew(nil, authContext)
+	sto.SetCacheDownloads(5)
```
Comment (bboozzoo) [inline overlord/overlord.go]: If I understand correctly, `5` is a default value. Can you add a const to hold this value?

File: store/cache.go [hunk 2]
```diff
@@ -85,6 +86,7 @@ func (cm *CacheManager) Get(cacheKey, targetPath string) error {
 	if err := os.Link(cm.path(cacheKey), targetPath); err != nil {
 		return err
 	}
+	logger.Debugf("CacheManager: using cache for %s", targetPath)
```
Comment (niemeyer) [inline store/cache.go]: The type name prefix sounds unnecessary and not very helpful for us or for users. Improving the message when unclear sounds better, but this message looks good on itself already.

File: store/store.go [hunk 3]
```diff
@@ -444,13 +437,15 @@ func New(cfg *Config, authContext auth.AuthContext) *Store {
 		detailFields:    fields,
 		authContext:     authContext,
 		deltaFormat:     deltaFormat,
-		cacher:          cacher,
+		cacher:          nil,
```
Comment (niemeyer) [inline store/store.go]: That's the default value. The line may be dropped.

File: store/store.go [hunk 4]
```diff
@@ -1977,3 +1972,16 @@ func (s *Store) ReadyToBuy(user *auth.UserState) error {
 		return &errors
 	}
 }
+
+func (s *Store) CacheDownloads() int {
+	return s.cfg.CacheDownloads
+}
+
+func (s *Store) SetCacheDownloads(amount int) {
```
Comment (niemeyer) [inline store/store.go]: Amount feels a bit ambiguous here. Amount of data or amount of files? What's the unit? nfiles or fileCount might be more clear.

File: store/store.go [hunk 5]
```diff
@@ -1977,3 +1972,16 @@ func (s *Store) ReadyToBuy(user *auth.UserState) error {
...
+	s.cfg.CacheDownloads = amount
+	if amount > 0 {
```
Comment (niemeyer) [inline store/store.go]: Bonus points for handling negatives correctly. :+1:

**Pull request merged**